src/github/workflow.ts

```diff
@@ -2,19 +2,27 @@ import { transformWorkflow } from "../transforms/transform-workflow";
 import { emitWebhookProcessedMetrics } from "utils/webhook-utils";
 import { createInstallationClient } from "utils/get-github-client-config";
 import { WebhookContext } from "../routes/github/webhook/webhook-context";
+import { getRepoConfig } from "services/user-config-service";
 
-export const workflowWebhookHandler = async (context: WebhookContext, jiraClient, _util, gitHubInstallationId: number): Promise<void> => {
+export const workflowWebhookHandler = async (context: WebhookContext, jiraClient, _util, gitHubInstallationId: number, subscription): Promise<void> => {
 	const { payload, log: logger } = context;
 	context.log = context.log.child({
 		jiraHost: jiraClient.baseURL,
 		gitHubInstallationId
 	});
+
+	const userConfig = await getRepoConfig(subscription, payload.repository.id);
+	if (payload.action == "requested" && userConfig && userConfig.serviceIds){
```

Comment:
When a workflow run is run for a protected environment, GitHub sends a `requested` event. From this event we can read the workflow ID, store it in the DB together with the service IDs from the user config file. For each of those associations we would have to run a scheduled job to check [the gating status](https://developer.atlassian.com/cloud/jira/software/rest/api-group-deployments/#api-deployments-0-1-pipelines-pipelineid-environments-environmentid-deployments-deploymentsequencenumber-gating-status-get)  regularly (we should stop polling after a given timeout so we don't overload our system). 

I don't think we can go without polling, after all, because the Jira API doesn't provide JSMs callback functionality (unless I missed something).   

When the status is approved, we call the [GitHub API](https://docs.github.com/en/rest/actions/workflow-runs#review-pending-deployments-for-a-workflow-run) to approve the pending deployment. To call this API, we may need to create a "robot" GitHub user which the user needs to add as "required reviewer" in their environment config.

Example workflow file: https://github.com/thombergs/deployments-test/blob/main/.github/workflows/main.yml

The payload of a `requested` webhook event looks like this:

```json
{
  "action": "requested",
  "workflow_run": {
    "id": 2901296991,
    "name": "My Deployment flow",
    "node_id": "WFR_kwLOHH2P2s6s7kdf",
    "head_branch": "main",
    "head_sha": "b14ac8f39038af071fcb25e9dbc131da323109e0",
    "path": ".github/workflows/main.yml",
    "run_number": 33,
    "event": "push",
    "status": "queued",
    "conclusion": null,
    "workflow_id": 23300142,
    "check_suite_id": 7914051586,
    "check_suite_node_id": "CS_kwDOHH2P2s8AAAAB17bYAg",
    "url": "https://api.github.com/repos/thombergs/deployments-test/actions/runs/2901296991",
    "html_url": "https://github.com/thombergs/deployments-test/actions/runs/2901296991",
    "pull_requests": [],
    "created_at": "2022-08-22T04:38:15Z",
    "updated_at": "2022-08-22T04:38:15Z",
    "actor": {
      "login": "thombergs",
      "id": 1363578,
      "node_id": "MDQ6VXNlcjEzNjM1Nzg=",
      "avatar_url": "https://avatars.githubusercontent.com/u/1363578?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/thombergs",
      "html_url": "https://github.com/thombergs",
      "followers_url": "https://api.github.com/users/thombergs/followers",
      "following_url": "https://api.github.com/users/thombergs/following{/other_user}",
      "gists_url": "https://api.github.com/users/thombergs/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/thombergs/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/thombergs/subscriptions",
      "organizations_url": "https://api.github.com/users/thombergs/orgs",
      "repos_url": "https://api.github.com/users/thombergs/repos",
      "events_url": "https://api.github.com/users/thombergs/events{/privacy}",
      "received_events_url": "https://api.github.com/users/thombergs/received_events",
      "type": "User",
      "site_admin": false
    },
    "run_attempt": 1,
    "referenced_workflows": [],
    "run_started_at": "2022-08-22T04:38:15Z",
    "triggering_actor": {
      ...
    },
   ...
      "author": {
        "name": "Tom Hombergs",
        "email": "tom.hombergs@gmail.com"
      },
      "committer": {
        "name": "GitHub",
        "email": "noreply@github.com"
      }
    },
    "repository": {
      "id": 477990874,
      "node_id": "R_kgDOHH2P2g",
      "name": "deployments-test",
      "full_name": "thombergs/deployments-test",
      "private": false,
      "owner": {
        "login": "thombergs",
        "id": 1363578,
      ...
      },
    ...
    },
    "head_repository": {
     ...
  },
  "repository": {
    "id": 477990874,
    "node_id": "R_kgDOHH2P2g",
    "name": "deployments-test",
    "full_name": "thombergs/deployments-test",
    "private": false,
    "owner": {
     ...
    },
    ...
  },
  "sender": {
   ...
  },
  "installation": {
    "id": 27957859,
    "node_id": "MDIzOkludGVncmF0aW9uSW5zdGFsbGF0aW9uMjc5NTc4NTk="
  }
}
```